### PR TITLE
🧪 Scout: improve sanitizeReturnTo test coverage and security

### PR DESCRIPTION
**Pull request by cungminh2710**
- 💡 What: Added security edge cases to `sanitizeReturnTo` tests and improved the implementation to handle URL-encoded and mixed-case paths.
- 🎯 Why: The previous implementation could be bypassed using encoding or casing, potentially leading to redirect loops or unauthorized redirects to sensitive auth routes.
- 🧩 Scope: `apps/hyperlocalise-web/src/lib/workos/return-to.ts` and `apps/hyperlocalise-web/src/lib/workos/return-to.test.ts`.
- ✅ Verification: Ran `cd apps/hyperlocalise-web && pnpm exec vp test src/lib/workos/return-to.test.ts`.
- 🛡️ Confidence: Hardens the application against Open Redirect-style bypasses and ensures robust redirection logic for authentication flows.

---
*PR created automatically by Jules for task [11297415136305770187](https://jules.google.com/task/11297415136305770187) started by @cungminh2710*

### CODE REVIEW

File: apps/hyperlocalise-web/src/lib/workos/return-to.test.ts [hunk 1]
```diff
@@ -52,4 +52,21 @@ describe("sanitizeReturnTo", () => {
     expect(sanitizeReturnTo("/not/auth/sign-in")).toBe("/not/auth/sign-in");
     expect(sanitizeReturnTo("/auth/sign-in-not-really")).toBe("/auth/sign-in-not-really");
   });
+
+  it("should detect URL-encoded restricted paths", () => {
+    // %73 is 's'
+    expect(sanitizeReturnTo("/auth/%73ign-in")).toBe("/dashboard");
+    // %2f is '/'
+    expect(sanitizeReturnTo("%2fauth/sign-in")).toBe("/dashboard");
+  });
```
[the remark this answers is not in the document]
Comment (cungminh2710) [reply apps/hyperlocalise-web/src/lib/workos/return-to.test.ts:61]: @jules fix

File: apps/hyperlocalise-web/src/lib/workos/return-to.ts [hunk 2]
```diff
@@ -9,13 +9,35 @@ const RESTRICTED_PATHS = [
...
 export function sanitizeReturnTo(value: string | null | undefined, fallback = "/dashboard") {
-  if (!value || !value.startsWith("/") || value.startsWith("//") || value.startsWith("/\\")) {
+  if (!value) {
+    return fallback;
+  }
+
+  // Normalise the value by decoding URI components and lowercasing for consistent checking.
+  let decodedValue: string;
+  try {
+    decodedValue = decodeURIComponent(value);
```
[the remark this answers is not in the document]
Comment (cungminh2710) [reply apps/hyperlocalise-web/src/lib/workos/return-to.ts]: @jules fix

File: apps/hyperlocalise-web/src/lib/workos/return-to.ts [hunk 3]
```diff
@@ -13,24 +13,37 @@ export function sanitizeReturnTo(value: string | null | undefined, fallback = "/
...
-    if (urlPath === "/auth/github/callback") {
-      const queryIndex = value.indexOf("?");
-      if (queryIndex !== -1) {
-        const params = new URLSearchParams(value.slice(queryIndex + 1));
-        if (params.has("installation_id") && params.has("state")) {
-          return value;
+    // Re-verify it still looks like a safe relative path after decoding.
+    if (urlPath.startsWith("//") || urlPath.startsWith("/\\")) {
+      return fallback;
+    }
```
[the remark this answers is not in the document]
Comment (cungminh2710) [reply apps/hyperlocalise-web/src/lib/workos/return-to.ts:24]: @jules fix
